.github/workflows/node.yml

```diff
@@ -0,0 +1,29 @@
+name: Node CI
+on: [push, pull_request]
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        node-version:
+          - 10.x
+        npm-script:
```

Comment:
This appears to be an abuse of the actions API. Matrix is intended to allow you to choose node and OS versions. If you want to run different kinds of tests, you should be defining multiple jobs.

Why do you need to define multiple jobs? ALL the jobs which you have listed here require the code generator to be run first. Only the jobs API supports the setup you need, allowing you to set up a sequential step, after which several different kinds of tasks are executed in parallel. That API is documented [here](https://help.github.com/en/articles/workflow-syntax-for-github-actions#jobsjob_idneeds)

Reply:
Very well, I will do as you wish when I'm back at my computer.

Reply:
> ALL the jobs which you have listed here require the code generator to be run first. Only the jobs API supports the setup you need, allowing you to set up a sequential step, after which several different kinds of tasks are executed in parallel.

I just discovered that is not how it works ([I tried on a different repo](https://github.com/ksxnodemodules/ts-pipe-compose/runs/228072115)): Each job run on their own virtual machine, completely isolated from each other, this is evidence by the fact that `runs_on` is required for every job. That means that if a "code generator" job already generate the code, other jobs won't have access to the generated code even if it depends on "code generator".

Reply:
So, do you still want to divide the matrix into multiple jobs?